Sources/Templates/AutoRegisterable.swift

```diff
@@ -23,35 +23,48 @@ enum AutoRegisterable {
             .filter(\.isAutoRegisterable)
 
         sortedProtocols.forEach { type in
-            let initMethods = type.methods.filter(\.isInitializer)
-            let registrationName = type.name.withLowercaseFirst().withoutLastCamelCasedPart()
-
-            // We only support up to 1 init method, otherwise we can't determine for what to generate the registration parameters.
-            if initMethods.count > 1 {
-                fatalError("Max 1 init method is supported for AutoRegisterable")
-            }
-
-            if let initMethod = initMethods.first {
-                let parameterType: String
-                // Factory works with tuples for more than 1 parameter.
-                if initMethod.parameters.count == 1 {
-                    // If we have a single parameter we directly set it.
-                    parameterType = initMethod.parameters[0].typeName.name
-                } else {
-                    // If there are more parameters we create a tuple with the types and argument labels
-                    // E.g. `(foo: String, bar: Int)`
-                    let joinedParameters = initMethod.parameters.map { "\($0.argumentLabel ?? $0.name): \($0.typeName.name)"}.joined(separator: ", ")
-                    parameterType = "(\(joinedParameters))"
+            if let registrationValues = type.registrationValues {
+                // If the registration values are specified we can use the regular `Factory`
+            	let nameAndValuePairs = registrationValues.components(separatedBy: ",")
+                nameAndValuePairs.forEach { pair in
+                    let nameAndValue = pair.components(separatedBy: "=")
+                    guard nameAndValue.count == 2 else { return }
+                    let registrationName = nameAndValue[0]
+                    lines.append("public var \(registrationName): Factory<\(type.name)> {".indent(level: 1))
```

Comment:
Perhaps we can move these 3 lines to a separate method? Then we can also use that for the same code from line 64-67

Reply:
Done ✔️ Also extracted adding the parameter factory lines to keep it consistent.